packages/SwingSet/src/kernel/vat-loader/manager-factory.js

```diff
@@ -54,66 +53,67 @@ export function makeVatManagerFactory({
     !setup || enableSetup || Fail`setup() provided, but not enabled`;
   }
 
-  // returns promise for new vatManager
-  function vatManagerFactory(
+  /**
+   * returns promise for new vatManager
+   *
+   * @param {import('../../types-internal.js').VatID} vatID
+   * @param {import('../../types-internal.js').ManagerOptions} managerOptions
+   * @param {import('@agoric/swingset-liveslots').LiveSlotsOptions} liveSlotsOptions
+   * @param {import('../vat-warehouse.js').VatSyscallHandler} vatSyscallHandler
+   * @returns { Promise<import('../../types-internal.js').VatManager> }
+   */
+  async function vatManagerFactory(
     vatID,
     managerOptions,
     liveSlotsOptions,
     vatSyscallHandler,
   ) {
```

Comment:
This function would also benefit from signature refactoring.
```suggestion
   * @param {import('../../types-internal.js').VatID} vatID
   * @param {import('../vat-warehouse.js').VatSyscallHandler} vatSyscallHandler
   * @param {object} options
   * @param {import('../../types-internal.js').ManagerOptions} options.managerOptions
   * @param {import('@agoric/swingset-liveslots').LiveSlotsOptions} options.liveSlotsOptions
   * @returns { Promise<import('../../types-internal.js').VatManager> }
   */
  async function vatManagerFactory(vatID, vatSyscallHandler, {
    managerOptions,
    liveSlotsOptions,
  }) {
```